packages/components/app/styles/components/table.scss

```diff
@@ -19,175 +19,166 @@ $hds-table-cell-padding-medium: 14px 16px 13px 16px; // the 1px difference is to
-  border-bottom: $hds-table-border-width solid $hds-table-border-color;
-  border-left: none;
-}
-
-.hds-table__th {
-  padding: $hds-table-cell-padding-medium;
-}
+    .hds-table__th--align-right,
+    .hds-table__td--align-right {
+      text-align: right;
 
-.hds-table__th-sort {
-  padding: 0;
-
-  button {
-    width: 100%;
-    height: 100%;
-    min-height: $hds-table-header-height;
-    margin: 0;
-    padding: $hds-table-cell-padding-medium;
-    text-align: inherit;
-    background-color: transparent;
-    border: 1px solid transparent;
-    border-radius: inherit;
-
-    .hds-table__th-sort--button-content {
-      display: flex; // we can't put flex on the button itself so we drop the content into its own container
-      align-items: center;
-
-      .flight-icon {
-        flex: none;
-        margin-left: 8px;
-        color: var(--token-color-foreground-action);
+      .hds-table__th-content {
+        justify-content: flex-end;
       }
     }
 
-    &:hover,
-    &.mock-hover {
-      color: var(--token-color-foreground-strong);
-      background-color: var(--token-color-palette-neutral-200);
-      cursor: pointer;
-    }
-
-    @include hds-focus-ring-with-pseudo-element($radius: inherit);
 
-    &:active,
-    &.mock-active {
-      color: var(--token-color-foreground-strong);
-      background-color: var(--token-color-palette-neutral-300);
+    .hds-table__th--align-center,
+    .hds-table__td--align-center {
+      text-align: center;
     }
-  }
-}
 
 
-// TABLE BODY
+    // border radius: target first and last th elements in the row
 
-// TABLE STRIPING ROWS
-.hds-table--striped {
-  .hds-table__tbody {
-    .hds-table__tr:nth-child(even) {
-      background-color: var(--token-color-surface-faint);
+    &:first-of-type {
+      th:first-child {
+        border-top-left-radius: $hds-table-inner-border-radius;
+      }
+
+      // last th in the row, border radius
+      th:last-child {
+        border-top-right-radius: $hds-table-inner-border-radius;
+      }
     }
   }
 }
 
-// TABLE DENSITY (TBODY ROW HEIGHT)
+.hds-table__th-content {
+  display: flex;
+  gap: 8px;
+  align-items: center;
+  justify-content: flex-start;
+}
 
-.hds-table__tbody th,
-.hds-table__tbody td {
-  .hds-table--density-short & {
-    padding: $hds-table-cell-padding-short;
-  }
+.hds-table__th-button {
+  display: flex;
+  flex: none;
+  align-items: center;
+  justify-content: center;
+  width: 24px;
+  height: 24px;
+  margin: -2px 0; // this is necessary to compensate for the height of the button being 24px while the line height of the text is 20px (the overall height of the cell shoud be 48px and we want to keep the cell's padding as is for consistency with Figma)
+  padding: 0;
+  color: var(--token-color-foreground-faint);
+  background-color: transparent;
+  border: 1px solid transparent;
+  border-radius: 3px;
 
-  .hds-table--density-medium & {
-    padding: $hds-table-cell-padding-medium;
+  &:hover,
+  &.mock-hover {
```

Comment:
The hover state for the sort/tooltip button mimics the `secondary` button and has `Elevation / Low` that only shows on hover.

Reply:
Done in https://github.com/hashicorp/design-system/pull/1860/commits/8bcb86e36fe11b140d5ccbd8d801d6426758621e